types/chat.ts

```diff
@@ -195,7 +213,32 @@ export interface FollowUpClickParams {
 
 /*
     Defines parameters for opening a tab.
-    Opens existing tab if `tabId` is provided, otherwise creates a new tab and opens it.
+    Opens existing tab if `tabId` is provided, otherwise creates a new tab
+    with options provided in `options` parameter and opens it.
 */
-export interface OpenTabParams extends Partial<TabEventParams> {}
+export interface OpenTabParams extends Partial<TabEventParams> {
+    newTabOptions?: {
+        state?: TabState
+        data?: TabData
+    }
+}
 export interface OpenTabResult extends TabEventParams {}
+
+export interface TabState {
+    inProgress?: boolean
+    cancellable?: boolean
```

Comment:
"cancellable" - practically means stop button will be shown
"inProgress" - it's being worked on by server, so no input should be accepted and progress bar shown

```typescript
/** Server is working, should show progress bar, should not accept input. */
inProgress?: boolean
/** Should show stop button. */
cancellable?: boolean
```